File: Ruby/Queries/PerfectNum.rb
def isPerfect(num)
    arr = []
    length = 0
    sum = 0
    (1...num).each do |i|
        if num % i == 0
            arr[length] = i
            length += 1
        end
    end
    (0...length).each do |i|
        sum = sum + arr[i]
    end
    if sum == num
        puts "Perfect num"
    else 
        puts "Not Perfect"
    end
end
puts "Enter num"
num = gets.to_i
isPerfect(num)

File: Ruby/Queries/PerfectNum.c
#include <stdio.h>
void isPerfect(int num){
    int i = 1, arr[100], lenght = 0,sum = 0;
    while (i < num){
        if(num % i == 0){
            arr[lenght] = i;
            lenght ++;
        }
        i++;
    }
    for(int i = 0; i < lenght; i++){
        sum = sum + arr[i];
    }
    if(sum == num){
        printf("Perfect num");
    }
    else {
        printf("Not Perfect");
    }
}
int main(){
    printf("Enter num \n");
    int num;
    scanf("%d", & num);
    isPerfect(num);
    return 0;
}